src/components/inputs/SignedIntInput.vue

```diff
@@ -0,0 +1,103 @@
+<template>
+<base-text-input
+    ref="input"
+    v-bind="$attrs"
+    :model-value="modelValue"
+    :label="shapedLabel"
+    :name="name"
+    :rules="rules"
+    :lazy-rules="false"
+    :size="undefined"
+    @update:modelValue="handleChange"
+/>
+</template>
+
+<script>
+import { BigNumber } from 'ethers';
+
+import { parseSignedIntString } from 'components/ContractTab/function-interface-utils';
+
+import BaseTextInput from 'components/inputs/BaseTextInput';
+
+export default {
+    name: 'SignedIntInput',
+    components: {
+        BaseTextInput,
+    },
+    emits: [
+        'update:modelValue',
+        'valueParsed',
+    ],
+    props: {
+        modelValue: {
+            type: String,
+            required: true,
+        },
+        label: {
+            type: String,
+            required: true,
+        },
+        name: {
+            type: String,
+            required: true,
+        },
+        // size in bits, in increments of 8 (i.e. bytes): // https://docs.soliditylang.org/en/latest/types.html#integers
+        // e.g. a size of 64 produces a type === int64
+        size: {
+            type: [Number, String],
+            required: true,
+            validator: size => {
+                return Number.isInteger(+size) &&
+                       +size % 8 === 0 &&
+                       +size <= 128;
```

Comment:
should move the byte check into an util method and refactor, see `SignedArrayIput.vue` component commented above.